pkg/virt-operator/install-strategy/create.go

```diff
@@ -366,185 +713,638 @@ func SyncAll(kv *v1.KubeVirt,
+			log.Log.V(2).Infof("serviceaccount %v created", sa.GetName())
+		} else if !objectMatchesVersion(&cachedSa.ObjectMeta, imageTag, imageRegistry) {
+			// Patch if old version
+			var ops []string
+
+			// Patch Labels and Annotations
+			labelAnnotationPatch, err := createLabelsAndAnnotationsPatch(&sa.ObjectMeta)
 			if err != nil {
-				return false, fmt.Errorf("unable to update role %+v: %v", r, err)
+				return err
 			}
-			log.Log.V(2).Infof("role %v updated", r.GetName())
+			ops = append(ops, labelAnnotationPatch...)
+
+			_, err = core.ServiceAccounts(kv.Namespace).Patch(sa.Name, types.JSONPatchType, generatePatchBytes(ops))
+			if err != nil {
+				return fmt.Errorf("unable to patch serviceaccount %+v: %v", sa, err)
+			}
+			log.Log.V(2).Infof("serviceaccount %v updated", sa.GetName())
 
 		} else {
-			log.Log.V(4).Infof("role %v already exists", r.GetName())
+			// Up to date
+			log.Log.V(2).Infof("serviceaccount %v already exists and is up-to-date", sa.GetName())
+		}
+	}
+
+	// create/update ClusterRoles
+	for _, cr := range targetStrategy.clusterRoles {
+		err := createOrUpdateClusterRole(cr,
+			kv,
+			imageTag,
+			imageRegistry,
+			stores,
+			clientset,
+			expectations)
+		if err != nil {
+			return err
+		}
+	}
+
+	// create/update ClusterRoleBindings
+	for _, crb := range targetStrategy.clusterRoleBindings {
+		err := createOrUpdateClusterRoleBinding(crb,
+			kv,
+			imageTag,
+			imageRegistry,
+			stores,
+			clientset,
+			expectations)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	// create/update Roles
+	for _, r := range targetStrategy.roles {
+		err := createOrUpdateRole(r,
+			kv,
+			imageTag,
+			imageRegistry,
+			kv.Namespace,
+			stores,
+			clientset,
+			expectations)
+		if err != nil {
+			return err
 		}
 	}
 
 	// create/update RoleBindings
 	for _, rb := range targetStrategy.roleBindings {
-		var cachedRb *rbacv1.RoleBinding
+		err := createOrUpdateRoleBinding(rb,
+			kv,
+			imageTag,
+			imageRegistry,
+			kv.Namespace,
+			stores,
+			clientset,
+			expectations)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
 
-		rb := rb.DeepCopy()
-		obj, exists, _ := stores.RoleBindingCache.Get(rb)
+// merges prev into target spec
+func mergeServiceSpec(targetSpec, prevSpec *corev1.ServiceSpec) *corev1.ServiceSpec {
+	mergedSpec := targetSpec.DeepCopy()
 
-		if exists {
-			cachedRb = obj.(*rbacv1.RoleBinding)
+	if prevSpec.Selector != nil {
+		if mergedSpec.Selector == nil {
+			mergedSpec.Selector = make(map[string]string)
 		}
 
-		injectOperatorLabelAndAnnotations(&rb.ObjectMeta, imageTag, imageRegistry)
-		if !exists {
-			// Create non existent
-			expectations.RoleBinding.RaiseExpectations(kvkey, 1, 0)
-			_, err := rbac.RoleBindings(kv.Namespace).Create(rb)
-			if err != nil {
-				expectations.RoleBinding.LowerExpectations(kvkey, 1, 0)
-				return false, fmt.Errorf("unable to create rolebinding %+v: %v", rb, err)
+		// merge any non-overlapping selectors
```

Comment:
The selectors are connected via `AND`, wouldn't we loose contact to all pods if we merge them?

Reply:
you're right... i think I've made this mistake in the past too. I keep thinking the labels are OR instead of AND. 

I'll remove all of this now.  The PR will be better for it anyway.

Reply:
fixed, i removed all of the service merging logic.  We either fully replace the Spec in the event of a type==ClusterIP, or delete and replace in the event of type!=ClusterIP.  I only feel comfortable patching type==ClusterIP because I've tested it in various conditions.  I'm not confident about which fields are mutable for other types.